src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,126 @@
+            statement.setLong(2, book.getPrice().longValue());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException(NO_ROWS_INSERTED_MSG);
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book = mapResultSetIntoBook(generatedKeys);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(CANNOT_CREATE_BOOK_MSG + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = connectionCreator.createConnection();
+                 PreparedStatement statement = connection.prepareStatement(SQL_SELECT_STATEMENT)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Book book = mapResultSetIntoBook(resultSet);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(CANNOT_FIND_OBJECT_MSG + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> list = new ArrayList<>();
+        try (Connection connection = connectionCreator.createConnection();
+                 PreparedStatement statement = connection.prepareStatement(
+                         SQL_SELECT_ALL_STATEMENT)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                list.add(mapResultSetIntoBook(resultSet));
+            }
+            return list;
+        } catch (SQLException e) {
+            throw new DataProcessingException(FAILED_TO_FIND_ROWS_MSG, e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = connectionCreator.createConnection();
+                 PreparedStatement statement = connection.prepareStatement(SQL_UPDATE_STATEMENT)) {
+            statement.setString(1, book.getTitle());
+            statement.setObject(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException(NO_NEEDED_ROW_MSG);
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException(CANNOT_UPDATE_ROW_MSG + book.getId(), e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = connectionCreator.createConnection();
+                 PreparedStatement statement = connection.prepareStatement(SQL_DELETE_STATEMENT)) {
+            statement.setLong(1, id);
+            int affectedRows = statement.executeUpdate();
+            return affectedRows >= 1;

```

Comment:
IMO it's more straightforward
```suggestion
            return statement.executeUpdate() > 0;
```